conformance/tests/admin-network-policy-core-gress-rules.go

```diff
@@ -0,0 +1,488 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tests
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"sigs.k8s.io/network-policy-api/apis/v1alpha1"
+	"sigs.k8s.io/network-policy-api/conformance/utils/kubernetes"
+	"sigs.k8s.io/network-policy-api/conformance/utils/suite"
+)
+
+func init() {
+	ConformanceTests = append(ConformanceTests,
+		AdminNetworkPolicyGress,
+	)
+}
+
+var AdminNetworkPolicyGress = suite.ConformanceTest{
+	ShortName:   "AdminNetworkPolicyGress",
+	Description: "Tests support for both ingress and egress traffic rules together using admin network policy API based on a server and client model",
+	Features: []suite.SupportedFeature{
+		suite.SupportAdminNetworkPolicy,
+	},
+	Manifests: []string{"tests/admin-network-policy-core-gress-rules_base.yaml"},
+	Test: func(t *testing.T, s *suite.ConformanceTestSuite) {
+
+		t.Run("Should support an 'allow-gress' policy across different protocols", func(t *testing.T) {
```

Comment:
This is really hard to review and I'm sure I'm missing things.... Maybe putting the explicit rule yaml each section is testing in a comment before it might help? (we can do this later)

Reply:
I feel like I need to see the yaml next to the test case

Reply:
yea i am going to move the yamls to the base folder like we discussed in previous PR and I think once this PR and next one lands I will spend some time writing docs that outline the design of these tests.